controllers/managedocs_controller.go

```diff
@@ -911,6 +916,26 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 	return nil
 }
 
+// reconcilePrometheusService function wait for prometheus Service
+// to start and sets appropriate annotation for 'service-ca' controller
+func (r *ManagedOCSReconciler) reconcilePrometheusService() error {
+	r.Log.Info("Reconciling Prometheus Service")
+	prometheusService := &corev1.Service{}
+	err := r.Client.Get(r.ctx,
+		types.NamespacedName{Name: prometheusServiceName, Namespace: r.namespace},
+		prometheusService)
+	// as this Service is not created explicitly by us,
+	// we don't raise any error, will wait till it is available
+	if err != nil {
+		return nil
+	}
```

Comment:
With the current changes, the above statement is inaccurate. 
You need to add a watch for the service in order for it to work (in the watch section at the start of this file)

In addition, not all errors are NotFound errors, and for other errors, you want to return the error so it will requeue another reconcile.

In my eyes it is also ok to just return an error for not found and not rely on a create event, worst case scenario you will only get one additional reconcile.

See last comment for the suggested return error

Reply:
Again this part is changed. Now we have our own `Service`.

Reply:
Same question as of the last comment: "Why do we need a secondary service if there is one that already exists? Or am I missing something?"